FirebaseDeviceStore/FirebaseDeviceStore.swift

```diff
@@ -123,8 +125,21 @@ public class FirebaseDeviceStore: NSObject, MessagingDelegate {
     private func deleteDevice(_ userId: String) {
         let docRef = userRef(userId);
 
+        // TODO: Rather than store a 'devices' array, you could store it as a map:
+        // devices: {
+        //   deviceId1: { },
+        //   deviceID2: { },
+        //   ...
+        // }
+        //
+        // And then to delete you don't need to use a transaction and you can just do
+        // docRef.updateData(FieldPath.of(self.DEVICES_FIELD, getDeviceId()), FieldValue.delete());
```

Comment:
I did plan on storing it as a map initially, but JS doesn't have the equivalent of a device ID, and as that was the first of the libraries I implemented, I ended up settling on an array.

In hindsight though, there's no reason that we couldn't use a pseudo device ID on the web and just have the complexity in a single library rather than across all three libraries.

I'll look to make this change on JS first to check I don't encounter any unforeseen issues and then port across to here and Android.